<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Groupbox */

QGroupBox {
    border: 1px solid #DDD;
    border-radius: 9px;
    margin-top: 9px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="OptionsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Options</string>
      </property>
      <widget class="QPushButton" name="LoadOBJ">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>181</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Load OBJ file</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QGroupBox" name="MeshGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>201</width>
         <height>161</height>
        </rect>
       </property>
       <property name="title">
        <string>Subdivision Settings</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="subdivTitleLabel">
          <property name="text">
           <string>Subdivision</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="stepsLayout">
          <item>
           <widget class="QLabel" name="SubdivLabel">
            <property name="text">
             <string>Steps:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="SubdivSteps">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QComboBox" name="MeshPresetComboBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>110</y>
         <width>181</width>
         <height>31</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>-</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Suzanne</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Spot</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>FancyCup</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>OpenCube</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>HollowTetra</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Brick</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>StackedCubesOpen</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>StackedCubesClosed</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RockerArm</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ItaPaperGeom</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Fertility</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Fandisk</string>
        </property>
       </item>
      </widget>
      <widget class="QGroupBox" name="renderSettingsGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>360</y>
         <width>201</width>
         <height>121</height>
        </rect>
       </property>
       <property name="title">
        <string>Render Settings</string>
       </property>
       <widget class="QCheckBox" name="limitProjectioncheckBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>161</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Limit projection</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="TessellationCheckBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>186</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>Tessellation</string>
        </property>
       </widget>
      </widget>
      <widget class="Line" name="line">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>160</y>
         <width>221</width>
         <height>20</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="tessSettingsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Tessellation Settings</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <widget class="QGroupBox" name="outerTessSettingsGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>201</width>
         <height>331</height>
        </rect>
       </property>
       <property name="title">
        <string>Outer tessellation levels</string>
       </property>
       <widget class="QLabel" name="outerTessLevel0Label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>171</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;For edge defined by  u=0:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="outerTessLevel0">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>70</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
       <widget class="QLabel" name="outerTessLevel1Label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>110</y>
          <width>171</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;For edge defined by v=0:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="outerTessLevel1">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>140</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
       <widget class="QLabel" name="outerTessLevel2Label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>180</y>
          <width>171</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;For edge defined by  u=1:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="outerTessLevel2">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>210</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
       <widget class="QLabel" name="outerTessLevel3Label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>250</y>
          <width>171</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;For edge defined by  v=1:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="outerTessLevel3">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>280</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
      </widget>
      <widget class="QCheckBox" name="HideMeshCheckBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>119</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Hide CPU Mesh</string>
       </property>
      </widget>
      <widget class="QLabel" name="chooseTessTypeLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>171</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Choose tessellation renderer&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QComboBox" name="tessTypecomboBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>171</width>
         <height>32</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>All patches</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Only regular patches</string>
        </property>
       </item>
      </widget>
      <widget class="QGroupBox" name="innerTessSettingsGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>540</y>
         <width>201</width>
         <height>181</height>
        </rect>
       </property>
       <property name="title">
        <string>Inner tessellation levels</string>
       </property>
       <widget class="QLabel" name="innerTess0Label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>191</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;For horizontal (u) edges:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="innerTessLevel0">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>70</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
       <widget class="QLabel" name="innerTess1Label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>110</y>
          <width>171</width>
          <height>30</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;For vertical (v) edges:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="innerTessLevel1">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>150</y>
          <width>161</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="MainDisplay">
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
